src/main/java/http/router/Router.java

```diff
@@ -1,5 +1,60 @@
 package http.router;
 
+
+import http.helper.Handler;
+import http.helper.Messages;
+import http.helper.Utils;
+import http.request.Request;
+import http.response.Response;
+
+import java.io.*;
+import java.net.Socket;
+import java.util.*;
+
+import static http.request.RequestParser.*;
+
 public class Router {
+ private Map<String, Route> routes = new HashMap<>();
+
+ public Map<String, Route> getRoutes(){
+     return routes;
+ }
+
+ public void makeRouterRequest (Socket clientSocket) throws IOException {
```

Comment:
This function does a lot - it seems to be a "management center' for the whole process - it reads the input, build the response, does the actual routing and sends the response. I wonder if it shouldn't be moved up, to the SocketHandler?

Reply:
I was hoping to abstract some stuff away but to a new class but maybe not socketHandler because it more so handles the response, request and handler classes. soo maybe a manager class or something like that.